src/pages/EditOrderPage.js

```diff
@@ -242,26 +269,34 @@ const EditOrderPage = () => {
   };
 
   const isElegibleForFreeDelivery = () => {
-    return hasDeliveryDiscount || (order.deliveryDiscount && !wasTakeAwayOrder()) || (cartTotal > order.amountForFreeDelivery && deliveryMethod === 'delivery');
+    return (hasDeliveryDiscount && !haveExtraFee) || (order.deliveryDiscount && !wasTakeAwayOrder()) || (cartTotal > order.amountForFreeDelivery && deliveryMethod === 'delivery' && !haveExtraFee);
   };
 
   const getDeliveryFee = () => {
```

Comment:
Break to another file

Reply:
This one the conditions are very different for each one of the 3 places i have it. It gets super confusing to have the logic for the 3 of them in one single function. I was thinking about keeping the way it is. But have a look and see if you find any better solution and let me know.

<img width="887" alt="Screenshot 2023-03-15 at 22 04 00" src="https://user-images.githubusercontent.com/69796068/225454071-0584de20-cebf-4fcb-a9cc-1e0c3372c0fa.png">